### ISSUE TITLE
box.backup may skip vylog file

### ISSUE THREAD

**Issue by kyukhin** [tarantool/tarantool#3624]
The issue can be reproduced on Tarantool 1.10 and 2.0. Tarantool 1.9 is unaffected.

Steps to reproduce:
1. Bootstrap a tarantool instance.
2. Restart the instance.
3. Create a vinyl space.
```
box.schema.space.create('test', {engine = 'vinyl'})
box.space.test:create_index('primary')
```
4. Make a snapshot.
```
box.snapshot()
```
5. Start backup.
```
box.backup.start()
```
It will return the following files to copy:
```
---
- - ./00000000000000000003.snap
...
```
That is there's no vylog file. An attempt to recover from this snapshot will result in an error:
```
2018-08-18 19:08:57.899 [18134] main/101/interactive F> can't initialize storage: Invalid VYLOG file: LSM tree 512/0 not found
```

The issue is occasionally caught by test-run:
```
[055] box/backup.test.lua
[055] Worker "055_box" received the following error; stopping...
[055] Traceback (most recent call last):
[055]   File "/home/kyukhin/w/tarantool/src/test-run/lib/worker.py", line 266, in run_task
[055]     task, self.server, self.inspector)
[055]   File "/home/kyukhin/w/tarantool/src/test-run/lib/test_suite.py", line 193, in run_test
[055]     short_status = test.run(server)
[055]   File "/home/kyukhin/w/tarantool/src/test-run/lib/test.py", line 166, in run
[055]     self.execute(server)
[055]   File "/home/kyukhin/w/tarantool/src/test-run/lib/tarantool_server.py", line 147, in execute
[055]     crash_occured = save_join(lua, timeout=self.TIMEOUT)
[055]   File "/home/kyukhin/w/tarantool/src/test-run/lib/tarantool_server.py", line 50, in save_join
[055]     green_obj.join(timeout=timeout)
[055]   File "/usr/lib64/python2.7/site-packages/gevent/greenlet.py", line 290, in join
[055]     result = self.parent.switch()
[055]   File "/usr/lib64/python2.7/site-packages/gevent/hub.py", line 331, in switch
[055]     return greenlet.switch(self)
[055] TarantoolStartError
[055]
[055]
[055]
[055] [Instance "backup_test" returns with non-zero exit code: 1]
[055]
[055] Last 15 lines of Tarantool Log file [Instance "backup_test"][/home/kyukhin/w/tarantool/src/test/var/055_box/backup_test.log]:
[055] Starting instance backup_test...
[055] Run console at unix/:/home/kyukhin/w/tarantool/src/test/var/055_box/backup_test.control
[055] started
[055] 2018-08-15 11:43:00.569 [17627] main/101/backup_test C> Tarantool 2.0.4-838-g32cb1ad
[055] 2018-08-15 11:43:00.569 [17627] main/101/backup_test C> log level 5
[055] 2018-08-15 11:43:00.570 [17627] main/101/backup_test I> mapping 268435456 bytes for memtx tuple arena...
[055] 2018-08-15 11:43:00.570 [17627] main/101/backup_test I> mapping 134217728 bytes for vinyl tuple arena...
[055] 2018-08-15 11:43:00.611 [17627] iproto/101/main I> binary: bound to [::1]:23850
[055] 2018-08-15 11:43:00.611 [17627] main/101/backup_test I> recovery start
[055] 2018-08-15 11:43:00.611 [17627] main/101/backup_test I> recovering from `/home/kyukhin/w/tarantool/src/test/var/055_box/backup_test/00000000000000000021.snap'
[055] 2018-08-15 11:43:00.651 [17627] main/101/backup_test vy_lsm.c:527 E> ER_INVALID_VYLOG_FILE: Invalid VYLOG file: LSM tree 514/0 not found
[055] 2018-08-15 11:43:00.651 [17627] main/101/backup_test F> can't initialize storage: Invalid VYLOG file: LSM tree 514/0 not found
[055] 2018-08-15 11:43:00.651 [17627] main/101/backup_test F> can't initialize storage: Invalid VYLOG file: LSM tree 514/0 not found
```

Here's where the test broke:
```
box.snapshot()
---
- ok
...
-- Copy files.
do_backup(files)
---
...
box.backup.stop()
---
...
-- Previous checkpoint must have been removed by garbage collection
-- as soon as box.backup.stop() was called.
files = box.backup.start(1) -- error: checkpoint not found
---
- error: Can't find snapshot
...
-- Check that we can restore from the backup we've just made.
_ = test_run:cmd(string.format("create server copy1 with script='box/backup_test.lua', workdir='%s'", BACKUP_DIR))
---
...
_ = test_run:cmd("start server copy1")
```

**Comment by locker** [comment]
Reproduce file:
```
- [box/access_bin.test.lua, null]
- [box/backup.test.lua, null]
```

**Comment by locker** [comment]
The culprit: 8e710090fcbc7506b0785ae90599eae38a363142

**Comment by locker** [comment]
Fixed by 2457be7115fe3a6557d64bab3191480b15a77844
